test/ClangImporter/AllowErrors/invalid-pch-bridging-header.swift

```diff
@@ -0,0 +1,29 @@
+// REQUIRES: objc_interop
+
+// RUN: %empty-directory(%t)
+// RUN: mkdir -p %t/pch %t/pch-dir
+// RUN: split-file %s %t
+
+// Check that the pch is output even though it has errors
+// RUN: %target-swift-frontend -emit-pch -o %t/pch/bridging-header.pch -Xcc -Xclang -Xcc -fallow-pcm-with-compiler-errors -Xcc -Xclang -Xcc -fmodule-format=raw %t/bridging-header.h
+// RUN: %target-swift-frontend -typecheck -verify -import-objc-header %t/pch/bridging-header.pch -Xcc -Xclang -Xcc -fallow-pcm-with-compiler-errors -Xcc -Xclang -Xcc -fmodule-format=raw %t/use.swift
+// RUN: ls %t/pch/*.pch | count 1
+
+// Same but with implicit PCH instead
+// RUN: %target-swift-frontend -typecheck -verify -import-objc-header %t/bridging-header.h -pch-output-dir %t/pch-dir -Xcc -Xclang -Xcc -fallow-pcm-with-compiler-errors -Xcc -Xclang -Xcc -fmodule-format=raw %t/use.swift
+// RUN: ls %t/pch-dir/*.pch | count 1
+
+// Second implicit run since we may go down a different path if the PCH already
+// exists
+// RUN: %target-swift-frontend -typecheck -verify -import-objc-header %t/bridging-header.h -pch-output-dir %t/pch-dir -Xcc -Xclang -Xcc -fallow-pcm-with-compiler-errors -Xcc -Xclang -Xcc -fmodule-format=raw %t/use.swift
+// RUN: ls %t/pch-dir/*.pch | count 1
+
+//--- bridging-header.h
+@import DoesNotExist;
+
+struct SomeTy {
+  int a;
+};
+
+//--- use.swift
+func use(s: SomeTy) {}
```

Comment:
I'm a little nervous about how thin these tests are. If I understand correctly, this change will allow invalid clang ASTs to appear in many parts of the ClangImporter where they were previously impossible, but your test case only exercises the tiny fraction of the importer that is involved in importing a struct without any interesting attributes or features, and there isn't actually anything invalid in this struct itself. Any other failure would not be caught here and would presumably surface as a weird crash in ClangImporter that we'd need to triage and test later.

Can you think of a way to (a) significantly improve test coverage (maybe you could re-run existing ClangImporter tests but with a broken code added to the headers?) and (b) make sure that crash logs clearly indicate that invalid clang ASTs were allowed?

Reply:
Invalid ASTs are already possible when `-fallow-pcm-with-compiler-errors` is passed in when loading PCMs (there's no check for diagnostics in that case). But I agree the test coverage isn't great here.

Regarding the crash logs, I had originally added a `While allowing modules with compiler errors enabled` line to the pretty stack trace a while ago, but have since removed that as it duplicates the information in the program arguments. I'm happy to put that back in if you think it'd be better to make it more obvious.

Reply:
I think it'd be good to add that back. The program arguments are often truncated and always difficult to visually parse.

Reply:
Unfortunately if the program arguments are truncated then the extra line won't be output anyway (since the arguments are the first thing printed). But I can add it in to address the second part (difficult to visually parse them).

Reply:
I've added that line back in as an extension to the existing frontend "Compiling with..."

If we think it's worth it I could also do some LLVM changes to allow printing the pretty stacktrace in reverse order. That would print arguments last rather than first (which then truncates everything). It currently reverses the stored "frames" in order to print, so should be pretty simple.